src/commands/payid-inspect.ts

```diff
@@ -0,0 +1,122 @@
+import * as Vorpal from 'vorpal'
+
+import { PaymentInformationInspector } from '../verifiable'
+import {
+  CertificateChainInspectionResult,
+  SignatureInspectionResult,
+} from '../verifiable/payment-information-inspector'
+
+import Command from './Command'
+import LocalStorage from './localstorage'
+
+/**
+ * Inspects the currently loaded PayID and prints inspection details to the console.
+ * Inspection looks at verified addresses and verifies the signatures and certificate chain (if present).
+ */
+export default class InspectPayIdCommand extends Command {
+  private readonly paymentInformationInspector: PaymentInformationInspector
+
+  /**
+   * Creates new command.
+   *
+   * @param vorpal - The vorpal instance to use.
+   * @param localStorage - The localstorage to use.
+   */
+  public constructor(vorpal: Vorpal, localStorage: LocalStorage) {
+    super(vorpal, localStorage)
+    this.paymentInformationInspector = new PaymentInformationInspector()
+  }
+
+  /**
+   * @override
+   */
+  protected command(): string {
+    return 'payid inspect'
+  }
+
+  /**
+   * @override
+   */
+  protected description(): string {
+    return 'Inspect signatures on the loaded PayID'
+  }
+
+  protected async action(): Promise<void> {
+    const info = this.getPaymentInfo()
+    const result = this.paymentInformationInspector.inspect(info)
+    this.vorpal.log(`${info.payId} ${validString(result.isVerified)}`)
+    result.verifiedAddressesResults.forEach((addressResult) => {
+      const address = addressResult.address
+      const cryptoAddress =
+        'address' in address.addressDetails
+          ? address.addressDetails.address
+          : ''
```

Comment:
```suggestion
      const cryptoAddress = address.addressDetails.address ?? ''
```
???

Reply:
I had to do it this way to make the linter happy. Because addressDetails has type `CryptoAddress | FiatAddress` and only CryptoAddress has an `address` property.

Reply:
What about `const cryptoAddress = address.addressDetails?.address ?? ''`?

If that doesn't work, then fine, I just dislike some nuances of TypeScript.